src/main/java/bio/terra/service/snapshotbuilder/utils/SearchConceptsQueryBuilder.java

```diff
@@ -51,12 +51,19 @@ public static String buildSearchConceptsQuery(
     var domainClause =
         createDomainClause(conceptTablePointer, conceptTableVariable, domainOption.getCategory());
 
+    List<FieldVariable> select = List.of(nameField, idField, countField);
+
+    List<TableVariable> tables = List.of(conceptTableVariable, domainOccurenceTableVariable);
+
+    List<OrderByVariable> orderBy =
+        List.of(new OrderByVariable(countField, OrderByDirection.DESCENDING));
+
+    List<FieldVariable> groupBy = List.of(nameField, idField);
+
     // if the search test is empty do not include the search clauses
     // return all concepts in the specified domain
     if (searchText == null || searchText.isEmpty()) {
-      // TODO: DC-845 Implement pagination, remove hardcoded limit
-      Query query =
-          new Query(List.of(nameField, idField), List.of(conceptTableVariable), domainClause, 100);
+      Query query = new Query(select, tables, domainClause, groupBy, orderBy, 100);
```

Comment:
This seems fine, but I wonder if there's a way to write this so there's only one call to `new Query();` and `query.renderSql()`. If the `if` condition was flipped, could the addition of the filter be made conditional?

Reply:
This is a good point, I added this `searchText.isEmpty() ? domainClause : where` so only one `new Query()` is made and one `query.renderSql()` is called. I would really like down the road to clean this function because there's a lot going on. Creating helper functions, adding stronger comments, ect.